gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/GenerateMetricSchemaTask.java

```diff
@@ -82,4 +98,10 @@ private String defaultLibraryName() {
         String rootProjectName = getProject().getRootProject().getName();
         return rootProjectName.replaceAll("-root$", "");
     }
+
+    private String defaultLibraryVersion() {
+        String version = Objects.toString(getProject().getRootProject().getVersion());
+        // Gradle returns 'unspecified' when there is no version information
+        return Strings.isNullOrEmpty(version) || "unspecified".equals(version) ? null : version;
+    }
```

Comment:
We could return null, falling back to the implementationVersion, when the version appears not to be a GA release or RC. That may be easier on caches, but I'm not thrilled with the idea of swapping the version tag implementation based on the presence of a tag. What do you think?

Reply:
Updated with this change